WORKFLOW.md

```diff
@@ -109,3 +109,40 @@ The project is usable as CMake subproject. Explicit references to
 break subproject builds.
 An example project is available [here](https://github.com/fortran-lang/stdlib-cmake-example)
 to test the CMake subproject integration.
+
+## GitHub collaboration
+
+Contributing can be daunting, we know! Even more for a big project with many contributors, and if you are not expert on the whole github workflow then even more, we have been there at some point.
+
+In order to help lowering the barrier for collaborating on ongoing efforts (e.g. an open PR), we have crafted a simple script that might come in handy. To explain the process we'll use Alice (the person you want to help) and Bob (you):
+
+       ┌────────────────────────────┐
+       │    fortran-lang/stdlib     │
+       └────────────▲───────────────┘
+                    │
+                    │  [Pull Request]
+                    │
+            ┌───────┴────────┐
+            │  alice/stdlib  │ ←─────┐
+            └──────▲─────────┘       │
+                   │                 │
+   [PR Branch] ←───┘          ┌──────┴──────┐
+   feature-branch             │  bob/stdlib │
+   (hosted here)              │    (fork)   │
+                              └─────────────┘
+                                     ▲
+                                     │
+                       [Push access to Alice's repo]
+
+After having forked and downloaded locally `stdlib` on your local machine; on an unix compatible terminal with access to the `git` CLI, being at the root folder: 
```

Comment:
I guess it should be mentioned that the fork is from `fortran-lang/stdlib` (and not from `alice/stdlib`)